quick-patched a crash

theRustyKnife opened the pull request:
When some tiles are placed at the signs position, the letters get
removed by the game since they are decorative. Then when the sign was
mined the mod would crash.
Some better solution like replacing the letters when they're destroyed
would be nice.

theRustyKnife committed:
quick-patched a crash

When some tiles are placed at the signs position, the letters get
removed by the game since they are decorative. Then when the sign was
mined the mod would crash.
Some better solution like replacing the letters when they're destroyed
would be nice.

Kasperki commented:
If we would change the letters from decorative to different entity type then they would not be removed?
Would entity type like wall or tree work? Then we would not need to functionality to replace them on destroyed.

theRustyKnife commented:
AFAIK the game only removes "decorative" entities when tiles are built. So if we were to change them to something else it should be ok, however they need the type to support "render_layer".

I haven't tested it, but from a quick look through the base prototypes it doesn't seem like neither "tree" nor "wall" does... From my search it seems like "simple-entity" could work (you can find it used in "Factorio\data\base\prototypes\entity\demo-doodads.lua" on line ~1716).
There's also "corpse", but I'm not sure how those work with the decay and stuff...

I can look into it more tomorrow if you don't have time, but these are my findings so far.

Cheers TRK

theRustyKnife committed:
Changed types to simple-entity, minor tweaks

All the types have been cahnged to simple-entity, letters and icons have
destructible set to false and empty collision mask.
Added event handlers for all the events when signs may be destroyed.
Tweaked icon size and position to prevent overlaping.
